node/controllers/auth/index.js

```diff
@@ -44,10 +55,76 @@ export const logout = (req, res) => {
   res.status(501).send('not yet implemented')
 }
 
-export const forgot = (req, res) => {
-  res.status(501).send('not yet implemented')
+export const forgot = async (req, res) => {
+  const { email } = req.body
+
+  return User.findOne({ where: { email } })
+    .then(async user => {
+      return PasswordResetUuid.create({ userId: user.id })
+    })
+    .then(async createdToken => {
+      const emailToSent = `Hei, meille tuli pyyntö resetoida salasanasi, tässä linkki josta pääset tekemään sen: 
+      \nwww.kohdataan.com/forgotPass?uuid=${createdToken.dataValues.uuid}`
+      await sendMail(email, 'PasswordResetLink', emailToSent)
+      res.status(201).send({
+        success: true,
+        message: 'Email found and uuid generated and stored',
+      })
+    })
+    .catch(err => {
+      res.status(500).send({
+        success: false,
+        message: 'Email not found',
+        error: err,
+      })
+    })
 }
 
-export const reset = (req, res) => {
-  res.status(501).send('not yet implemented')
+export const reset = async (req, res) => {
+  const { uuid, password } = req.body
+
+  return PasswordResetUuid.findOne({ where: { uuid } })
```

Comment:
Looks good to me, but just out of curiosity: Is there something related to models that need promise-chaining instead of just awaiting everything? Sequalize seems to support async/await, so why not just use that instead of promises and async functions together? I believe this works fine like this, but for me it always looks a bit weird if there are promise chains and async functions mixed together. Catch is of course handy for error handling throughout the entire promise chain, so maybe that's one reason?

Reply:
Main reason was that i was going to use only promise chaining like other controllers using sequelize models, but the easiest way to update mattermost user and send the email is with the async/await. I can refactor the forgot and reset -functions not to use promise chaining pretty easily thought if needed